src/v2/cookbook/form-validation.md

```diff
@@ -0,0 +1,322 @@
+---
+title: Form Validation
+type: cookbook
+order: 1.2
+---
+
+## Simple Example
+
+Form validation is natively supported by the browser, but sometimes different browsers will handle things in a manner which makes relying on it a bit tricky. Even when validation is supported perfectly, there may be times when custom validations are needed and a more manual, Vue-based solution may be more appropriate. Let's begin with a simple example.
+
+Given a form of three fields, make two required. Let's look at the HTML first:
+
+``` html
+<form id="app" @submit="checkForm" action="/something" method="post">
+  
+  <p v-if="errors.length">
+    <b>Please correct the following error(s):</b>
+    <ul>
+      <li v-for="error in errors">{{ error }}</li>
+    </ul>
+  </p>
+  
+  <p>
+    <label for="name">Name<label>
+    <input type="text" name="name" id="name" v-model="name">
+  </p>
+
+  <p>
+    <label for="age">Age<label>
+    <input type="number" name="age" id="age" v-model="age">
+  </p>
+
+  <p>
+    <label for="movie">Favorite Movie<label>
+    <select name="movie" id="movie" v-model="movie">
+      <option>Star Wars</option>
+      <option>Vanilla Sky</option>
+      <option>Atomic Blonde</option>
+    </select>
+  </p>
+
+  <p>
+    <input type="submit" value="Submit">  
+  </p>
+
+</form>
+```
+
+Let's cover it from the top. The form tag has an ID that we'll be using for the Vue component. There's a submit handler that you'll see in a bit, and the action is a fake URL that would point to something real on a server someplace (where you have backup server-side validation of course).
+
+Beneath that there is a paragraph that shows or hides itself based on an error state. This will render a simple list of errors on top of the form. Also note we fire the validation on submit rather than as every field is modified. 
+
+The final thing to note is that each of the three fields has a corresponding v-model to connect them to values we will work with in the JavaScript. Now let's look at that.
+
+``` js
+const app = new Vue({
+  el:'#app',
+  data:{
+    errors:[],
+    name:null,
+    age:null,
+    movie:null
+  },
+  methods:{
+    checkForm:function(e) {
+      if(this.name && this.age) return true;
+      this.errors = [];
+      if(!this.name) this.errors.push("Name required.");
+      if(!this.age) this.errors.push("Age required.");
+      e.preventDefault();
+    }
+  }
+})
+```
+
+Fairly short and simple. We default an array to hold errors and set null values for the three form fields. The checkForm logic (which is run on submit remember) checks for name and age only as movie is optional. If they are empty we check each and set a specific error for each. And that's really it. You can run the demo below. Don't forget that on a successful submission it's going to POST to a non-existent URL.

```

Comment:
The weird part about this demo is that it posts to nothing, so playing with the demo will give the user a blank page. You kind of address this in the last sentence, but I would either add a success to the demo, or make the last sentence a warning, and say specifically that the POST URL will go to success page that they will have to build. (nitpicky, I know, but right now the demo feels a little funny.

Reply:
I'm a bit unsure how to handle this. I worry that if I post to a real URL, like Vue's home page, that would be confusing as well. Could I make the last sentence stronger? Like maybe just bold it? And perhaps add an exclamation mark? And then maybe we can see if reader's get confused by itself and send us feedback?

Reply:
It will only update within the iframe of the embed, so that would be ok. Can you please try it out and if that doesn't work we can go back and do bold or something.

Reply:
Do we want to change the embeds and text? If both, I'll update, if just the embeds, you can do that on your end, right?

Reply:
Disregard previous comment.